From 74944430b987da6c71a089f066d5fedbb8a710c0 Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Fri, 10 Jun 2016 09:09:24 +0700
Subject: [PATCH] DoorPhone-Support auto recognize screen on RZG1EM and iWave

At current time, DoorPhone chooses screen #1 (usually HDMI) to get information
so DoorPhone will error if screen #1 is not available

This patch file modifies DoorPhone for it auto recognize the plugged screen on the
board, in case there are 2 plugged screens, DoorPhone will choose the screen that has
higher resolution (as larger screen is usually default screen for demo)

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 main.cpp | 15 +++++++++++++--
 1 file changed, 12 insertions(+), 1 deletions(-)

diff --git a/main.cpp b/main.cpp
index f70950d..18bd5f1 100755
--- a/main.cpp
+++ b/main.cpp
@@ -65,7 +65,18 @@ int main(int argc, char *argv[])
     p_app = &app;	/* For calling quit in signal handler */
 
     //Show information of screen (all monitors)
-    QScreen *screen = QGuiApplication::screens().at(1);
+    // If 2 screen availabe, chose the larger as it is usually default
+    QScreen *screen, *tmp_screen;
+    if(QGuiApplication::screens().at(0) == NULL)
+		screen = QGuiApplication::screens().at(1);
+	  else if(QGuiApplication::screens().at(1) == NULL)
+		screen = QGuiApplication::screens().at(0);
+	  else {
+		screen = QGuiApplication::screens().at(0);
+		tmp_screen = QGuiApplication::screens().at(1);
+		if((screen->availableSize().width() * screen->availableSize().height()) < (tmp_screen->availableSize().width() * tmp_screen->availableSize().height()))
+			screen = QGuiApplication::screens().at(1);
+	  }
     qDebug() << "Information for screen:" << screen->name();
     qDebug() << "  Available geometry:" << screen->availableGeometry().x() << screen->availableGeometry().y() << screen->availableGeometry().width() << "x" << screen->availableGeometry().height();
     qDebug() << "  Available size:" << screen->availableSize().width() << "x" << screen->availableSize().height();
-- 
1.9.1

